### ISSUE TITLE
generation logits dtype bug

### ISSUE THREAD

**Issue by binhtranmcs** [triton-inference-server/tensorrtllm_backend#598]
### System Info

- GPU: rtx4090
- Nvidia driver: 535.86.10
- Ubuntu 22.04.4

### Who can help?

@byshiue @schetlur-nv 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Same as [here](https://github.com/NVIDIA/TensorRT-LLM/issues/2088).

### Expected behavior

correct logits are returned to client.

### actual behavior

logits are all wrong.

### additional notes

After further debugging, I notice that the bug comes from [here](https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/inflight_batcher_llm/src/model_instance_state.cc#L897). The `result.generationLogits` tensor can have dtype float16, but the backend assumes that it is float32. So the `generationLogitsBuffer` contains the wrong values.

I can do a for loop to convert each element to float32, but I think it is not an optimal solution. Is there a good way to handle this in general? Please have a look. Thanks in advance.

**Comment by ShuaiShao93** [comment]
I got the same issue. Any updates?

**Comment by ShuaiShao93** [comment]
The python backend seems to work tho: https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/all_models/inflight_batcher_llm/tensorrt_llm/1/model.py

**Comment by ShuaiShao93** [comment]
Ok, the best solution is actually setting `--logits_dtype=float32` in trtllm-build
